### ISSUE TITLE
Send and forget messages

### ISSUE THREAD

**Issue by vdaele** [obgm/libcoap#256]
After sending a number (3 to 4) of messages to an **unreachable** light to the Ikea Hub, 
I notice that subsequent requests to change the colors of **reachable** lights don't seem to get through:
however I do get responses acknowledging that the light was changed but the color of the bulb itself didn't change.
I tried to improve this by using NON messages but that didn't help (so I guess the problem is not located in the libcoap - hub communication).

Are there other ways to tell to the server that it can forget requests when they can't be sent?

I also tried the following option, alas, also without effect

```
        coap_insert_optlist(&optlist, coap_new_optlist(COAP_OPTION_NORESPONSE, 0, NULL));
```

I assumed that `COAP_OPTION_NORESPONSE` would imply that the server would not send responses anymore (but it still does).  Is my assumption correct or do I incorrectly set this option?

Looking forward to your feedback,

Marc

**Comment by obgm** [comment]
The No-Response option tells a server that the sender does not care about the response and the server therefore does not need to send it. (Which makes most sense for non-confirmable requests as otherwise, the server would have to send the acknowledgment anyway.)
I suppose, your Ikea hub does not support the No-Response option and therefore keeps sending the 2.xx responses. Which is ok as No-Response is an elective CoAP option.

Maybe the initial requests somehow eat up resources in the Ikea hub's Zigbee implementation so that the new requests are processed on the CoAP side but not on the Zigbee side?

**Comment by vdaele** [comment]
Thanks for your input.  Seems a plausible explanation.

Does the code itself, for sending the COAP_OPTION_NORESPONSE, look ok?
Is it correct thet sending 0x2 is correct if I'm not interested in 2.xx messages?

**Comment by obgm** [comment]
Yes, this looks correct to me, including 0x2 as value for suppressing 2.xx responses only.
